shell/platform/android/io/flutter/embedding/engine/loader/ResourceExtractor.java

```diff
@@ -21,111 +20,63 @@
-
-      final String timestamp = checkTimestamp(dataDir, mPackageManager, mPackageName);
-      if (timestamp == null) {
-        return null;
-      }
-
-      deleteFiles(mDataDirPath, mResources);
-
-      if (!extractAPK(dataDir)) {
-        return null;
-      }
-
-      if (timestamp != null) {
-        try {
-          new File(dataDir, timestamp).createNewFile();
-        } catch (IOException e) {
-          Log.w(TAG, "Failed to write resource timestamp");
-        }
-      }
-
-      return null;
-    }
-
-    /// Returns true if successfully unpacked APK resources,
-    /// otherwise deletes all resources and returns false.
-    @WorkerThread
-    private boolean extractAPK(@NonNull File dataDir) {
-      for (String asset : mResources) {
-        try {
-          final String resource = "assets/" + asset;
-          final File output = new File(dataDir, asset);
-          if (output.exists()) {
-            continue;
-          }
-          if (output.getParentFile() != null) {
-            output.getParentFile().mkdirs();
-          }
-
-          try (InputStream is = mAssetManager.open(asset);
-              OutputStream os = new FileOutputStream(output)) {
-            copy(is, os);
-          }
-          if (BuildConfig.DEBUG) {
-            Log.i(TAG, "Extracted baseline resource " + resource);
-          }
-        } catch (FileNotFoundException fnfe) {
-          continue;
-
-        } catch (IOException ioe) {
-          Log.w(TAG, "Exception unpacking resources: " + ioe.getMessage());
-          deleteFiles(mDataDirPath, mResources);
-          return false;
-        }
-      }
-
-      return true;
-    }
-  }
-
   @NonNull private final String mDataDirPath;
   @NonNull private final String mPackageName;
   @NonNull private final PackageManager mPackageManager;
   @NonNull private final AssetManager mAssetManager;
   @NonNull private final HashSet<String> mResources;
-  private ExtractTask mExtractTask;
+  private FutureTask mExtractTask;
+
+  private static final ThreadFactory mThreadFactory =
+      new ThreadFactory() {
+        private final AtomicInteger mCount = new AtomicInteger(1);
+
+        public Thread newThread(Runnable r) {
+          return new Thread(r, TAG + " #" + mCount.getAndIncrement());
+        }
+      };
+
+  // Used only for rejected executions.
+  private static ThreadPoolExecutor mBackupExecutor;
+  private static LinkedBlockingQueue<Runnable> mBackupExecutorQueue;
+
+  private static final RejectedExecutionHandler mRunOnSerialPolicy =
+      new RejectedExecutionHandler() {
+        public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
+          Log.w(TAG, "Exceeded ThreadPoolExecutor pool size");
+          synchronized (this) {
+            if (mBackupExecutor == null) {
+              mBackupExecutorQueue = new LinkedBlockingQueue<Runnable>();
+              mBackupExecutor =
+                  new ThreadPoolExecutor(
+                      5 /* BACK UP POOL SIZE */,
```

Comment:
Same here, what led to these settings for the thread pool?

Reply:
Same thing here!